### PR TITLE
Run Bundlemon on release optimized build

### PR DESCRIPTION
**Pull request by jonkoops**
This should resolve the issues we found under #7905, namely the size check showing up twice. Also runs Bundlemon against a release optimized build to ensure that the file size is reported accurately.

### CODE REVIEW

**Review by mourner — APPROVED** [review]
Nice! Although I don't fully understand why we need a separate workflow to compare production builds — can you explain? I thought removing duplication is fixed solely by the config.

Also, nit but I wonder why we see those small +/- ~30 bytes changes in the check when the bundle itself shouldn't change.

**Review by mourner — CHANGES_REQUESTED** [review]
Per author's comment, let's keep it in the main workflow — it seems separating it is not necessary after all.

**Comment by jonkoops** [comment]
> Also, nit but I wonder why we see those small +/- ~30 bytes changes in the check when the bundle itself shouldn't change.

This is because we are currently running a development build on the `main` branch. This build will always add GIT information such as the branch and short ref (see [here](https://github.com/Leaflet/Leaflet/blob/main/build/rollup-config.js#L12)). This is the primary reason why I've put the code in a separate workflow.

I think I could make it so that the main workflow also does a production build, let me experiment a little.

**Comment by mourner** [comment]
> This build will always add GIT information such as the branch and short ref (see [here](https://github.com/Leaflet/Leaflet/blob/main/build/rollup-config.js#L12)).

Ah, I got confused by the terminology — it's still a production build (it's minified, etc.), just not a release build. Hashes are fine I think, after all both branch and `main` builds should have them. We could remove branch name from the version name to keep them the same length, but also not a big deal if there are small differences shown by the checks.

**Comment by Falke-Design** [comment]
I see the project is built by the workflow, is it possible to get access to the `dist` folder? To create demos with the new leaflet-src?

**Comment by jonkoops** [comment]
@mourner My bad, yeah I meant the release optimized build in this case, which does not include any GIT information.

I split the build out into two distinct steps now, one for a release build and one for a regular build. Bundlemon will now always compare the file size of the release build (both on `main` as the PR) so that no file size changes can sneak in from the build system.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 name: CI
-on: [push, pull_request]
+on:
+  push:
+    branches: [main]
+  pull_request:
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply .github/workflows/main.yml]: Agreed

**Pull request merged**